plugins/graphql-voyager/README.md

```diff
@@ -0,0 +1,75 @@
+# graphql-voyager
+
+Welcome to the graphql-voyager plugin!
+
+_This plugin was created through the Backstage CLI_
+
+## Getting started
+
+### Installing
+
+To get started, first install the plugin with the following command:
+
+```bash
+# From your Backstage root directory
+yarn add --cwd packages/app @backstage/plugin-graphql-voyager
+```
+
+### Adding the page
+
+In order to be able to navigate to the graphQL voyager page, a new route needs to be added to the app. This can be done in the following way:
+
+```tsx
+// packages/app/App.tsx
+
+  import { GraphQLVoyagerPage } from "@backstage/plugin-graphql-voyager";
+
+  const routes = (
+    <FlatRoutes>
+      <Route path="/graphql-voyager" element={<GraphqlVoyagerPage/>}/>
+```
+
+### Configuration
+
+In order for the plugin to function correctly. GraphQL endpoints need to be added / configured through the GraphQLVoyager API. This can be done by implementing the `graphQlVoyagerApiRef` exported by this plugin.
+
+Add your own configuration to the `packages/app/src/apis.ts` file the following way:
+
+```ts
+import { identityApiRef } from './IdentityApi';
+import { GraphQLVoyagerEndpoints } from './GraphQLVoyagerEndpoints';
+
+export const apis: AnyApiFactory[] = [
+  createApiFactory({
+    api: graphQlVoyagerApiRef,
+    deps: { identityApi: identityApiRef },
+    factory: ({ identityApiRef }) => {
+      GraphQLVoyagerEndpoints.from([
+        {
+          id: `graphql-voyager-endpoint-id`,
+          title: 'endpoint-title',
+          introspectionErrorMessage:
+            'Unable to perform introspection, make sure you are on the correct environment.',
+          introspection: async (query: any) => {
+            const token = 'someSecretJWTComingFromIdentityApiRef';
+
+            const res = await fetch('graphQLEndpoint', {
+              method: 'POST',
+              body: JSON.stringify({ query }),
+              headers: {
+                'Content-Type': 'application/json',
+                Authorization: token,
+              },
+            });
+
+            return res.json();
```

Comment:
One thing that we commonly do, is to use [this little stanza](https://github.com/backstage/backstage/blob/086a557cd5d62a01eb4739e23bce32fd296ddfec/STYLE.md?plain=1#L47) that wraps the error into a standard shape if the request failed. The good thing is, it'll understand when the upstream is a Backstage service that responds with the standardized serialized error shapes. Then you get tracebacks and all kinds of detailed info, and can display them with a `ResponseErrorPanel` in the frontend.

I know this is just example code and probably want to keep it short. Just wanted to mention anyway :)